yew-ui/src/model/media_devices/media_device_access.rs

```diff
@@ -0,0 +1,62 @@
+use gloo_utils::window;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::JsFuture;
+use web_sys::MediaStreamConstraints;
+use yew::prelude::Callback;
+
+pub struct MediaDeviceAccess {
+    granted: Arc<AtomicBool>,
+    pub on_granted: Callback<()>,
+    pub on_denied: Callback<()>,
+}
+
+impl MediaDeviceAccess {
+    pub fn new() -> Self {
+        Self {
+            granted: Arc::new(AtomicBool::new(false)),
+            on_granted: Callback::noop(),
+            on_denied: Callback::noop(),
+        }
+    }
+
+    pub fn is_granted(&self) -> bool {
+        self.granted.load(Ordering::Acquire)
+    }
+
+    pub fn request(&self) {
+        let future = Self::request_permissions();
+        let on_granted = self.on_granted.clone();
+        let on_denied = self.on_granted.clone();
```

Comment:
This should be:

``` let on_denied = self.on_denied.clone();```

Reply:
d'oh!   embarrassing.    (i had that tested and working earlier and somehow must have "cleaned up" the code and broken it.)

good catch.